plugin/core/session_view.py

```diff
@@ -0,0 +1,189 @@
+from .logging import debug
+from .protocol import TextDocumentSyncKindNone, TextDocumentSyncKindFull
+from .sessions import Session
+from .settings import settings
+from .types import view2scope
+from .typing import Any, Iterable, Optional
+from .views import did_change
+from .views import did_close
+from .views import did_open
+from .views import did_save
+from .views import will_save
+from .windows import AbstractViewListener
+from .windows import debounced
+import sublime
+import weakref
+
+
+class PendingBuffer:
+
+    __slots__ = ('version', 'changes')
+
+    def __init__(self, version: int, changes: Iterable[sublime.TextChange]) -> None:
+        self.version = version
+        self.changes = list(changes)
+
+    def update(self, version: int, changes: Iterable[sublime.TextChange]) -> None:
+        self.version = version
+        self.changes.extend(changes)
+
+
+class SessionView:
+
+    def __init__(self, listener: AbstractViewListener, session: Session) -> None:
+        self.view = listener.view
+        self.session = session
+        self.listener = weakref.ref(listener)
+        self._file_name = self.view.file_name() or ""
```

Comment:
We do have to store the filename (and the buffer ID, and the view ID), because the `__del__` method of SessionView is usually called later than the `__del__` of ViewListener.
Moreover, we also need to store the file name in case of a rename of the file.

Reply:
I saw that another place uses it to handle renaming but not this class. So I think that that argument doesn't apply here.

If we need it then let's use it everywhere, also in `on_text_changed`.